Найти:
1.	Общее число оценок в файле
2.	Общее количество пользователей, поставивших оценки
3.	Общее количество оцененных фильмов
4.	ID самого активного пользователя
5.	Фильм, собравший наибольшее количество оценок


Анализ задания: 
Для выполнения задания необходимо произвести анализ данных, содержащихся в csv файле. Для этого целесообразно применить инструмент pandas.
Ход решения задачи можно продумать после предварительного анализа данных.

Выполним загрузку файла csv в DataFrame pandas.

In [215]:
import pandas as pd
#Загружаем файл
df = pd.read_csv("./ratings_small.csv")

Оценим структуру таблицы и распрделение значений.

In [216]:
print(df)

        userId  movieId  rating   timestamp
0            1       31     2.5  1260759144
1            1     1029     3.0  1260759179
2            1     1061     3.0  1260759182
3            1     1129     2.0  1260759185
4            1     1172     4.0  1260759205
...        ...      ...     ...         ...
99999      671     6268     2.5  1065579370
100000     671     6269     4.0  1065149201
100001     671     6365     4.0  1070940363
100002     671     6385     2.5  1070979663
100003     671     6565     3.5  1074784724

[100004 rows x 4 columns]


In [217]:
df.describe()

userId        movieId         rating     timestamp
count  100004.000000  100004.000000  100004.000000  1.000040e+05
mean      347.011310   12548.664363       3.543608  1.129639e+09
std       195.163838   26369.198969       1.058064  1.916858e+08
min         1.000000       1.000000       0.500000  7.896520e+08
25%       182.000000    1028.000000       3.000000  9.658478e+08
50%       367.000000    2406.500000       4.000000  1.110422e+09
75%       520.000000    5418.000000       4.000000  1.296192e+09
max       671.000000  163949.000000       5.000000  1.476641e+09

1. Общее число оценок в файле это количество записей.
2. Общее количество пользователей, поставивших оценки - максимальное значение столбца UserID
3. Судя по максимальному значению movieId, превышающему количество записей в таблице, большое количество возможных значений id не используется, при этом значения повторяются.
Количество уникальных значений в movieId будут количеством оцененных фильмов.
4. Чтобы найти ID самого активного пользователя необходиммо найти уникальные значения userId и их количество в таблице и выбрать элемент с максимальным количеством записей.
5. Чтобы найти Id фильма с наибольшим количеством оценок необходиммо найти уникальные значения movieId и их количество в таблице и выбрать элемент с максимальным количеством записей.


In [218]:
print ("Число оценок =", df.last_valid_index()+1) 

Число оценок = 100004


In [219]:
# Найдем максимальное значение столбца UserID

print ("Общее количество пользователей, поставивших оценки =", df['userId'].max())

Общее количество пользователей, поставивших оценки = 671


In [220]:
# Найдем уникальные значения movieId и посчитаем их количество
unique_count = df['movieId'].nunique()
print("Общее количество оцененных фильмов:", unique_count)

Общее количество оцененных фильмов: 9066


Создадим объект Series с уникальными значениями userId и их количеством в таблице и отсортируем его.
Самое большое значение - ID самого активного пользователя.

In [221]:
count_user = df['userId'].value_counts().sort_values()
print ("ID самого активного пользователя:",count_user.index[-1])

ID самого активного пользователя: 547


Создадим объект Series с уникальными значениями movieId и их количеством в таблице и отсортируем его.
Самое большое значение - Id фильма с наибольшим количеством оценок.

In [224]:
count_values = df['movieId'].value_counts().sort_values()
print ("ID фильма с наибольшим количеством оценок:",count_values.index[-1])

ID фильма с наибольшим количеством оценок: 356


Выводы:
1. Общее количество оценок в таблице совпадает с количеством записей.
2. Количество пользователей, поставивших оценки - 671 человек. Значения находятся в диапазоне от 1 до 671.
3. Значения movieId не являются непрерывным списком значений, о чем говорит их распределение.